src/game.cpp

```diff
@@ -11762,16 +11762,33 @@ void game::update_overmap_seen()
 {
     const tripoint ompos = u.global_omt_location();
     const int dist = u.overmap_sight_range( light_level( u.posz() ) );
+    const int dist_squared = dist * dist;
     // We can always see where we're standing
     overmap_buffer.set_seen( ompos.x, ompos.y, ompos.z, true );
-    for( int x = ompos.x - dist; x <= ompos.x + dist; x++ ) {
-        for( int y = ompos.y - dist; y <= ompos.y + dist; y++ ) {
+    for( int dx = -dist; dx <= dist; dx++ ) {
+        for( int dy = -dist; dy <= dist; dy++ ) {
+            if( trigdist && dx * dx + dy * dy > dist_squared ) {
+                continue;
+            }
+            int x = ompos.x + dx;
+            int y = ompos.y + dy;
+            float multiplier;
+            if( trigdist ) {
+                float angle = atan2( y - ompos.y, x - ompos.x ) + M_PI;
+                angle = fmod( angle, M_PI / 2 );
```

Comment:
```suggestion
               float angle = fmod( atan2( y - ompos.y, x - ompos.x ) + M_PI, M_PI / 2 );
```
And remove the upper line with declaration of `angle`.